### ISSUE TITLE
Generalize formatters which working through pipe

### ISSUE THREAD

**Issue by actionless** [google/vim-codefmt#5]

**Comment by dbarnett** [comment]
Github doesn't let me add you as an assignee, but we'll consider it assigned. =)

What did you have in mind? I'd suggest starting with a script-local helper for the formatters in codefmt. There are some hacks like passing just the range to the formatter that don't work reliably and I'm nervous about things like that getting baked into the API, but it's definitely useful to refactor and see what strong patterns arise.

**Comment by actionless** [comment]
yup, i meant organizing repeatable blocks from formatters' code into small functions
